### DISCUSSION TITLE
Activity correlation

### DISCUSSION THREAD

**Discussion by rbsis** [danielgerlag/workflow-core#1276] [Q&A]
When starting multiple Activity workflows of the same type (with the same workflow Id and activity name), how can the pending activity be found for one of the specific workflows? GetPendingActivity appears the only way to get an activyty, however it returns the latest instead of a specific one.

**Comment by ftdbowler** [comment]
same question... GetPendingActivity seems to simply return first... but if you had a bunch of activities... perhaps for approval process steps, you would very much want it to map to a particular workflow.

**Reply by ftdbowler** [reply]
I found answer in the ActivityName expression that incorporates workflowid... Works great!

    class ActivityWorkflow : IWorkflow<MyData>
    {
        public string Id => "activity-sample";
        public int Version => 1;

        public void Build(IWorkflowBuilder<MyData> builder)
        {
            builder
                .StartWith<HelloWorld>()
                .Activity((data, context) => "get-approval-" + context.Workflow.Id, (data) => data.Request)
                    .Output(data => data.ApprovedBy, step => step.Result)
                .Then<CustomMessage>()
                    .Input(step => step.Message, data => "Approved by " + data.ApprovedBy)
                .Then<GoodbyeWorld>();
        }
    }
